mne_bids/mne_bids.py

```diff
@@ -256,30 +253,26 @@ def _participants_tsv(raw, subject_id, group, fname, overwrite=False,
         else:
             subject_age = "n/a"
 
-        data.update({'age': [subject_age], 'sex': [sex], 'group': [group]})
-
-    df = pd.DataFrame(data=data,
-                      columns=['participant_id', 'age', 'sex', 'group'])
+    df = OrderedDict([('participant_id', [subject_id]),
+                      ('age', [subject_age]), ('sex', [sex]),
+                      ('group', [group])])
 
     if os.path.exists(fname):
-        orig_df = pd.read_csv(fname, sep='\t')
-        # whether the data exists identically in the current DataFrame
-        exact_included = df.values.tolist()[0] in orig_df.values.tolist()
-        # whether the subject id is in the existing DataFrame
-        sid_included = subject_id in orig_df['participant_id'].values
+        orig_df = from_tsv(fname)
+        # whether the data exists identically in the current MockDataFrame
+        exact_included = contains_row(orig_df,
+                                      [subject_id, subject_age, sex, group])
+        # whether the subject id is in the existing MockDataFrame
+        sid_included = subject_id in orig_df['participant_id']
         # if the subject data provided is different to the currently existing
         # data and overwrite is not True raise an error
         if (sid_included and not exact_included) and not overwrite:
             raise OSError(errno.EEXIST, '"%s" already exists in the '
                           'participant list. Please set overwrite to '
                           'True.' % subject_id)
         # otherwise add the new data
-        df = orig_df.append(df)
-        # and drop any duplicates as we want overwrite = True to force the old
-        # data to be overwritten
-        df.drop_duplicates(subset='participant_id', keep='last',
-                           inplace=True)
-        df = df.sort_values(by='participant_id')
+        combine(orig_df, df, 'participant_id')
+        df = orig_df
```

Comment:
my feeling is that you need to rework the logic here because we don't use dataframes any more. Having two different functions for `combine` and `contains_row` is probably making it more complicated than simple. Could your "`df`" ever have more than one row? Maybe that's increasing your complexity ...

fight complexity as much as you can :)